lua/neo-tree/ui/renderer.lua

```diff
@@ -17,6 +17,27 @@ local default_popup_size = { width = 60, height = "80%" }
 local floating_windows = {}
 local draw, create_window, create_tree, render_tree
 
+-- clean up neotree buffers created by :mksession
+local cleaned_up = false
+local clean_neotree_buffers = function()
+  if cleaned_up then
+    return
+  end
+  local clean_buffer_with_prefix = function(prefix)
+    for bufnr = 1, vim.fn.bufnr("$") do
```

Comment:
I think this may create errors if one of the numbers in the range is not a valid buffer. This code would be a better way to iterate over all buffers:
```
  for _, buf in ipairs(vim.api.nvim_list_bufs()) do
    local buf_name = vim.api.nvim_buf_get_name(buf)
```